### PR TITLE
Skip reaction notifications for bookmarks

### LINKED ISSUE #19356: Save (bookmark) does not create proper notification
**Describe the bug**

When an article is saved/bookmarked by a user, the author will not receive a "bookmark" notification. Instead, they will receive a previous "like" notification.

**To Reproduce**
1. reader clicks save on an article.
2. author receives a notification.
3. author will not realize it's not supposed to be a "like" notification, but supposed to be a "save".

**Expected behavior**

Author should receive a "save" notification.

**Screenshots**
n/a

**Additional context**

### PR DESCRIPTION
**Pull request by filleduchaos**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We only really intend to send notifications for proper reactions (fire, unicorn, etc), but since bookmarks are internally modelled as reactions they have been inadvertently triggering reaction notifications as well. This PR introduces a `notifiable` list of `ReactionCategory`s so that we can control which reactions or reaction-like things cause a notification to be sent out.

## Related Tickets & Documents

- Related Issue #19356 
- Closes #19356 

Note: The product part of this issue (notifications for bookmarks specifically) has been split off into another issue (#19510) so for now, removing bookmarks from the notifications for reactions should be enough to close the original issue.

## QA Instructions, Screenshots, Recordings

https://www.loom.com/share/940a281ec18c4263ae4a589db4f411ef

### UI accessibility concerns?

N/A

## Added/updated tests?

- [x] Yes, updated the reaction handler specs
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?

Nope

### CODE REVIEW

**Comment by mirie** [comment]
Flagging @pkfrank since I put [Save (bookmark) does not create proper notification](https://github.com/forem/forem/issues/19356) in `blocked` and `needs product` status. Do we want to send notification when a user bookmarks an author's article?

**Comment by maestromac** [comment]
Marking this as draft for product discussion.

**Comment by filleduchaos** [comment]
Sorry - please note updated PR description, the product work has been moved to a separate issue so this is now just a bug fix :)

File: app/services/reaction_handler.rb [hunk 1]
```diff
@@ -146,13 +146,17 @@ def sink_articles(reaction)
...
+    return unless ReactionCategory.notifiable.include?(reaction.category.to_sym)
+
     Notification.send_reaction_notification(reaction, reaction.target_user)
     return unless reaction.reaction_on_organization_article?
 
     Notification.send_reaction_notification(reaction, reaction.reactable.organization)
   end
 
   def send_notifications_without_delay(reaction)
+    return unless ReactionCategory.notifiable.include?(reaction.category.to_sym)
```
Comment (jaw6) [inline app/services/reaction_handler.rb:158]: This is reaching into two dependencies in a relatively deep way — and twice in this class. What about adding some method like `Reaction#should_notify?` that encapsulates the logic?

**Review by jaw6 — CHANGES_REQUESTED** [review]

File: app/models/reaction.rb [hunk 2]
```diff
@@ -139,6 +139,7 @@ def contradictory_mod_reactions(category:, reactable_id:, reactable_type:, user:
   # - reaction is negative
   # - receiver is the same user as the one who reacted
   # - reaction status is marked invalid
+  # - reaction is not in a category that should be notified
   def skip_notification_for?(_receiver)
```
Comment (jaw6) [inline app/models/reaction.rb:143]: At some point in the future, we ought to refactor this method: `_receiver` is required but isn't used, which also makes the method name and explanatory comment a bit misleading.

**Review by jaw6 — APPROVED** [review]
:+1:

**Review by maestromac — APPROVED** [review]
LGTM!

**Pull request merged**